ui/lib/CreateStream.tsx

```diff
@@ -75,58 +111,120 @@ const CreateStream = (props: CreateStreamProps) => {
+      props.registerStream
     );
   };
 
-  return (
-    <div id="CreateStream">
-      <div>
-        {/* Display: What is the product they are paying for? With what token? */}
-        {/* @ts-ignore */}
-        {props.cart && props.cart}
-      </div>
+  // Set transaction amount
+  const transactionAmount = Number(
+    (
+      (selectedTime * props.amountPerSecond) /
+      Math.pow(10, tokenData?.decimals || 0)
+    ).toFixed(Math.min(tokenData?.decimals || 0, 3))
+  );
 
-      {/* Part 1: The user needs to decide how long they want the stream to run for  */}
-      <div id="CreateStream-lifetime">
-        <Slider
-          min={1}
-          marks={marks}
-          max={maxTimeDays}
-          step={null}
-          value={Math.min(selectedTime / SECS_PER_DAY, maxTimeDays)}
-          defaultValue={1}
-          onChange={(value) =>
-            typeof value === "number" && setSelectedTime(SECS_PER_DAY * value)
-          }
-        />
-      </div>
-      <br />
-
-      {/* Part 2: The user approves exactly X tokens, corresponding to stream life   */}
-      <div id="CreateStream-approve">
-        <button disabled={!approveStream} onClick={() => approveStream?.()}>
-          {`Approve ${(
-            (selectedTime * props.amountPerSecond) /
-            Math.pow(10, tokenData?.decimals || 0)
-          ).toFixed(
-            Math.min(tokenData?.decimals || 0, 3),
-          )} ${tokenData?.symbol}`}
-        </button>
-      </div>
-      <br />
-
-      {/* Part 3: The user creates their stream, using the given reason/product code */}
-      <div id="CreateStream-create">
-        <button onClick={createStream}>
-          {`Open Stream for ${selectedTime / SECS_PER_DAY} day${
-            selectedTime !== SECS_PER_DAY ? "s" : ""
-          }`}
-        </button>
+  // Open stream button only if Approve button has been clicked
+  const [isApproved, setIsApproved] = useState(false);
+  const onApproveClick = () => {
+    approveStream?.();
+    setIsApproved(true);
+  };
+
+  const renderBalanceCheck = () => {
+    // Step 1: Check gas and native token balance
+    if (gasPrice === null || balance === null) {
+      return (
+        <div id="CreateStream">
+          <p>(Step 1/2) Checking gas and native token balance...</p>
+        </div>
+      );
+    }
+
+    // Step 2: Check if there are enough native tokens for fees
+    if (gasPrice >= balance) {
+      return (
+        <div id="CreateStream">
+          <p> Not enough native tokens to pay for transaction fees</p>
+          <button
+            onClick={(e) => {
+              window.location.href = "https://hop.exchange/";
+            }}
+          >
+            Go to Hop Exchange
+          </button>
+        </div>
+      );
+    }
+
+    // Step 3: Check stream token balance
+    if (transactionAmount === null || tokenBalance === null) {
+      return (
+        <div id="CreateStream">
+          <p>(Step 2/2) Checking stream token balance...</p>
```

Comment:
This way it just falls-through if this sequence finishes first (no confusion about "what was step 2/2?")
```suggestion
          <p>Checking stream token balance...</p>
```